concrete-core/src/backends/core/implementation/engines/glwe_ciphertext_consuming_retrieval.rs

```diff
@@ -0,0 +1,295 @@
+use crate::backends::core::implementation::engines::CoreEngine;
+use crate::backends::core::implementation::entities::{
+    GlweCiphertext32, GlweCiphertext64, GlweCiphertextMutView32, GlweCiphertextMutView64,
+    GlweCiphertextView32, GlweCiphertextView64,
+};
+use crate::backends::core::private::math::tensor::IntoTensor;
+use crate::specification::engines::{
+    GlweCiphertextConsumingRetrievalEngine, GlweCiphertextRetrievalError,
+};
+
+/// # Description:
+/// Implementation of [`LweCiphertextConsumingRetrievalEngine`] for [`CoreEngine`] that returns the
+/// underlying vec of a [`GlweCiphertext32`] consuming it in the process
+impl GlweCiphertextConsumingRetrievalEngine<GlweCiphertext32, Vec<u32>> for CoreEngine {
+    /// # Example:
+    /// ```
+    /// use concrete_commons::parameters::{GlweSize, PolynomialSize};
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // Here we create a container outside of the engine
+    /// // Note that the size here is just for demonstration purposes and should not be chosen
+    /// // without proper security analysis for production
+    /// let glwe_size = GlweSize(600);
+    /// let polynomial_size = PolynomialSize(1024);
+    ///
+    /// // You have to make sure you size the container properly
+    /// let mut owned_container = vec![0_u32; glwe_size.0 * polynomial_size.0];
+    /// let original_vec_ptr = owned_container.as_ptr();
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let ciphertext: GlweCiphertext32 =
+    ///     engine.create_glwe_ciphertext(owned_container, polynomial_size)?;
```

Comment:
Well I wanted to show we get the actual exact container, I think I did it well for the mut slice case with pointers, not sure it's clear for vec based containers

Reply:
Vec containers are checked with a pointer as well, immutable slices are compared directly as it seems to check the underlying pointer directly. Mutable slices are checked with a pointer as it's not allowed to take two mutable slices to the same memory

Reply:
Let me know if you think encryption is better, but to show we retrieve the actual exact container this would be the way to go (as it differs from classic retrieval that have existed up to now)

Reply:
Ok sounds good to me!